### DISCUSSION TITLE
Schedule Schema Definition

### DISCUSSION THREAD

**Discussion by ChakshuGautam** [beckn/DSEP-Specification#89] [Ideas]
## Use Cases
1. Lecture Schedule to showcase the lecture details and when they are getting scheduled. The schedule can be a list of ivites imported directly to my calendar to schedule or compare.

<img width="1427" alt="Screenshot 2022-12-15 at 12 49 55 PM" src="https://user-images.githubusercontent.com/7413816/207797558-31e69ec6-d98e-4ce4-95a2-3ee62219edf5.png">

## Solution
Wiki [defines](https://en.wikipedia.org/wiki/Schedule) a schedule as "_A schedule or a timetable, as a basic time-management tool, consists of a list of times at which possible tasks, events, or actions are intended to take place._" The current definition of `times` includes the definition of `data-time` but doesn't include the actual metadata of what will happen in that `event`.

It is proposed to extend the schema to include `events` as a first-class citizen in the Schedule schema as follows.

```yaml
Schedule:
  description: Describes a schedule
  type: object
  properties:
    events:
      type: array
      items:
        type: object
        properties:
          description:
            $ref: '#/components/schemas/Descriptor'
          time:
            $ref: '#/components/schemas/Time'
    frequency:
      $ref: '#/components/schemas/Duration'
    holidays:
      type: array
      items:
        type: string
        format: date-time
    times:
      type: array
      items:
        type: string
        format: date-time
```

This is keeping in mind that `times` is continued for backward compatibility.

PR here - #90

**Comment by emmayank** [comment]
Hii @ravi-prakash-v , can you please take a look here and through some light to the above suggestion by @ChakshuGautam
